rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/45_knn_search_bit.yml

```diff
@@ -354,3 +354,54 @@ setup:
               dims: 40
               index: true
               similarity: max_inner_product
+
+
+---
+"Search with synthetic source":
+  - requires:
+      capabilities:
+        - method: POST
+          path: /_search
+          capabilities: [ bit_dense_vector_synthetic_source ]
+      test_runner_features: capabilities
+      reason: "Support for bit dense vector synthetic source capability required"
+  - do:
+      indices.create:
+        index: test_synthetic_source
+        body:
+          mappings:
+            properties:
+              name:
+                type: keyword
+              vector1:
+                type: dense_vector
+                element_type: bit
+                dims: 40
+                index: false
+              vector2:
+                type: dense_vector
+                element_type: bit
+                dims: 40
+                index: true
+                similarity: l2_norm
+
+  - do:
+      index:
+        index: test_synthetic_source
+        id: "1"
+        body:
+          name: cow.jpg
+          vector1: [2, -1, 1, 4, -3]
+          vector2: [2, -1, 1, 4, -3]
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
+        force_synthetic_source: true
+        index: test_synthetic_source
+
+  - match: {hits.hits.0._id: "1"}
+  - match: {hits.hits.0._source.vector1: [2, -1, 1, 4, -3]}
+  - match: {hits.hits.0._source.vector2: [2, -1, 1, 4, -3]}
```

Comment:
In addition to this test, could you add some coverage to DenseVectorFieldMapperTests? Something under `DenseVectorSyntheticSourceSupport` ?

Reply:
Of course, I'll look into it.

Reply:
@benwtrent 
Thank you for your patience. I have added tests for the BIT type in DenseVectorSyntheticSourceSupport. However, I seem to have discovered another bug. During the parsing process, since it is done sequentially, the `element_type` is not yet parsed when `dims` is being processed. As a result, the default value is FLOAT instead of BIT, causing the `dims` range to be [1, 4096] instead of [8, 32768].
![debug](https://github.com/user-attachments/assets/c8804224-8339-45ec-9b72-ea5e61492f3b)

Here is a test to illustrate the issue:

```
PUT test
{
  "mappings": {
    "properties": {
      "vector": {
        "type": "dense_vector",
        "dims": 8000,
        "index": false,
        "element_type": "bit"
      }
    }
  }
}

{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "The number of dimensions for field [vector] should be in the range [1, 4096] but was [8000]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Failed to parse mapping: The number of dimensions for field [vector] should be in the range [1, 4096] but was [8000]",
    "caused_by": {
      "type": "mapper_parsing_exception",
      "reason": "The number of dimensions for field [vector] should be in the range [1, 4096] but was [8000]"
    }
  },
  "status": 400
}
```

This behavior contradicts the expected 32768 dimensions for the BIT type.

To avoid test failures, I have temporarily limited the `dims` in the DenseVectorSyntheticSourceSupport tests to 512 (4096 / 8) when using the BIT type. 

I'm not sure how to proceed from here. Should we address this issue in the current PR, or handle it separately? Any advice would be appreciated.